client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-plugin-section.tsx

```diff
@@ -91,27 +91,36 @@ export const ThankYouPluginSection = ( { plugin }: { plugin: any } ) => {
 		( state ) => isFetchingUserPurchases( state ) || ! hasLoadedUserPurchasesFromServer( state )
 	);
 	useQueryUserPurchases();
+	const [ expirationDate, setExpirationDate ] = useState( '' );
 
-	const productPurchase = purchases?.find( ( item ) => {
-		if ( item.siteId === siteId ) {
-			if (
-				plugin.variations?.monthly?.product_id === item.productId ||
-				plugin.variations?.yearly?.product_id === item.productId
-			) {
-				return item;
-			}
-		}
-	} );
+	const productPurchase = useMemo(
```

Comment:
As we talked @WBerredo, currently `plugin` object still gets too much changes, so this code will help us in the future when we have those improvements done.

Reply:
I think this function is similar to https://github.com/Automattic/wp-calypso/blob/a14b6727ffc1ed68b951c95f8a43ca6e1cb15ccc/client/lib/plugins/utils.js#L457, would it make sense consolidating them?

Reply:
I just made a change to that function in https://github.com/Automattic/wp-calypso/pull/74462

Reply:
Yes, let me push a commit for this!

Reply:
I have just merged https://github.com/Automattic/wp-calypso/pull/74462, and you can remove the `isMarketplaceProduct`